### PR TITLE
Exam portal

### PR DESCRIPTION
**Pull request by himanshu1365**

### CODE REVIEW

File: ExamPortal/Models/examDetail.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const mongoose  = require('mongoose')
+
+const Schema = mongoose.Schema
+
+const examSchema = new Schema({
+	examName: String,
+	examCode: String,
+	facultyId: String,
```
Comment (techfreakworm) [inline ExamPortal/Models/examDetail.js]: What does facultyId mean here? If it's not the correct key name, please change it. It's very misleading.

File: ExamPortal/Models/question.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		op2: String,
+		op3: String,
+		op4: String
+	},
+	weightage: Number,
+	time: {
+		type: Date,
+		default: Date.now
+	},
+	examCode: String
```
Comment (techfreakworm) [inline ExamPortal/Models/question.js]: It should be `_id` of examSchema, which indeed would be an ObjectID

File: ExamPortal/Models/testRecord.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const mongoose  = require('mongoose')
+
+const Schema = mongoose.Schema
+
+const testSchema = new Schema({
```
Comment (techfreakworm) [inline ExamPortal/Models/testRecord.js]: rename the schema to `candidateAnswer` or suggest a better name, `testSchema` is misleading.

File: ExamPortal/Models/examDetail.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	instructions: String,
+	examDuration: String
+})
+
+const exam = mongoose.model('examSchema',examSchema)
```
Comment (techfreakworm) [inline ExamPortal/Models/examDetail.js]: Should be `exam` instead of `examSchema`

File: ExamPortal/Models/question.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+})
+
+const question = mongoose.model('questionSchema',questionSchema)
```
Comment (techfreakworm) [inline ExamPortal/Models/question.js]: Should be `question` instead of `questionSchema`

File: ExamPortal/Models/testRecord.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	},
+	answers: [
+		{
+			answerSubmitted : String,
+			questionId: String
+		}
+	]
+})
+
+const test = mongoose.model('testSchema',testSchema)
```
Comment (techfreakworm) [inline ExamPortal/Models/testRecord.js]: should be `test` instead of `testSchema`

File: ExamPortal/Models/testRecord.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+module.exports = {
+	test
+}
```
Comment (techfreakworm) [inline ExamPortal/Models/testRecord.js]: Enter a newline here.

File: ExamPortal/Models/userRecord.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+	collegeName: String,
+	phoneNumber: String,
+	createdAt: {
+		type: Date,
+		default: Date.now
+	},
+	permissionLevel: Number
+})
+
+const user = mongoose.model('userSchema',userSchema)
```
Comment (techfreakworm) [inline ExamPortal/Models/userRecord.js]: should be `user` instead of `userSchema`

File: ExamPortal/Models/userRecord.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+const mongoose  = require('mongoose')
+
+const Schema = mongoose.Schema
+
+const userSchema = new Schema({
+	email: String,
+	name: String,
+	password:String,
+	accType: String,
```
Comment (techfreakworm) [inline ExamPortal/Models/userRecord.js]: make it `accountType`

File: ExamPortal/Models/question.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
+const mongoose  = require('mongoose')
+
+const Schema = mongoose.Schema
+
+const questionSchema = new Schema({
+	qHeading: String,
```
Comment (techfreakworm) [inline ExamPortal/Models/question.js]: make it `questionHeading`

File: ExamPortal/Models/question.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+	qText: String,
```
Comment (techfreakworm) [inline ExamPortal/Models/question.js]: similarly `questionText`

File: ExamPortal/Models/question.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+	qNo: Number,
```
Comment (techfreakworm) [inline ExamPortal/Models/question.js]: What exactly is this? Will it decide the sequence of the question. If so, will user enter the sequence of the question?

File: ExamPortal/Models/question.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+	ans: String,
+	options:{
+		op1: String,
```
Comment (techfreakworm) [inline ExamPortal/Models/question.js]: `option1`

File: ExamPortal/Models/question.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+		op2: String,
+		op3: String,
```
Comment (techfreakworm) [inline ExamPortal/Models/question.js]: `option3`

File: ExamPortal/Models/question.js [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+		op4: String
```
Comment (techfreakworm) [inline ExamPortal/Models/question.js]: `option4`

File: ExamPortal/Models/question.js (hunk 2 above)
Comment (birendra-bit) [reply ExamPortal/Models/question.js]: We have created a unique exam code which will be test key for student. we discussed it with Vishal too.
Birendra here

**Pull request merged**